src/servers/src/postgres/auth_handler.rs

```diff
@@ -23,11 +23,31 @@ use pgwire::error::{ErrorInfo, PgWireError, PgWireResult};
 use pgwire::messages::response::ErrorResponse;
 use pgwire::messages::startup::Authentication;
 use pgwire::messages::{PgWireBackendMessage, PgWireFrontendMessage};
+use snafu::ResultExt;
 
-struct PgPwdVerifier;
+use crate::auth::{Identity, Password, UserProviderRef};
+use crate::error;
+use crate::error::Result;
+
+struct PgPwdVerifier {
+    user_provider: Option<UserProviderRef>,
+}
 
 impl PgPwdVerifier {
-    async fn verify_pwd(&self, _pwd: &str, _meta: HashMap<String, String>) -> PgWireResult<bool> {
+    async fn verify_pwd(&self, pwd: &str, meta: HashMap<String, String>) -> Result<bool> {
+        let user_name = match meta.get("user") {
+            Some(name) => name,
+            None => return Ok(false),
```

Comment:
shouldn't return ok if `UserProvider` is provided. in other words, if `UserProvider` is provided, it's forced to check password. 

check `if let Some(user_provider) = &self.user_provider` at the beginning of every auth process